### PR TITLE
Feature/add clickable snackbar notification

### PR DESCRIPTION
**Pull request by oktapodia**
I'm not confident about reselect and had a lot of problem to get only one transaction, I decided to take the `transactions` like the `HistoryPage` and do a `find` on it.

- Each transaction can be accessible via `/transactions/:txHash`
- The `TxDetails` page can be accessible during the pending state
- The snackbar transaction is now clickable

### CODE REVIEW

**Comment by go1dfish** [comment]
Haven't tested this yet, but the code looks good.

Reselect is not actually used for the most part because the redux state gets mutated in ways it shouldn't, but once I move the selectors over to it your code here should still be fine.

Like the approach of just searching the transaction list for now as that avoids extra rpc work.  Ideally I'd like to move that find to the selectors code instead of the component (they get passed (state, ownProps) and each function in selectorMap({..}) gets passed the same.  But I don't think that should block this merge.

File: app/components/views/TransactionPage/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        {...{
+          ...this.props,
+          ...this.state,
+          transactionDetails: this.getTransaction(),
+        }}
+      />
+    );
+  }
+
+  getTransaction() {
```
Comment (go1dfish) [inline app/components/views/TransactionPage/index.js]: If this gets moved to a selector we can probably eliminate this component entirely.

something like:

    viewedTransaction = createSelector(
      [transactions, (state, { params: { txHash }}) => txHash]
      (transactions, txHash) => find(transactions.All, { txHash })
    )

The find pulled into selectors is the functional one though so you will probably use find({ txHash }, transactions.All) instead.

**Comment by alexlyp** [comment]
Over all, works pretty well.  Though noticed a couple things:

1) It doesn't quite respond to my click as I'd expect, he to kind of chase it to fire the view change.
2) Now that this is essentially revealing pending transactions details pages there is some special casing that needs to be done to reflect that it is an unmined transaction.  Should be pretty clear what needs fixing in there.

**Comment by oktapodia** [comment]
Updated the transaction details view when the transaction is in pending to:
- Hide the block section
- Hide the timestamp (upright)
- Set confirmation to 0

**Pull request merged**